File: Python/color_balls/color_balls.py

# Create a dictionary to store balls
balls = {}

# Read input and store it in dictionary
line = input()
while line != "End":
    color, count = line.split(': ')
    balls[color] = int(count)
    line = input()

min_color = ""
min_count = float("inf")

# Find the smallest color group
for color, count in balls.items():
    if count < min_count:
        min_count = count
        min_color = color

# Take all balls except for the smallest color group
min_sum = 0
for color, count in balls.items():
    if color != min_color:
        min_sum += count

print(f"Minimum number of balls: {min_sum}")

File: Java/src/ColorBalls.java
import java.util.Map;
import java.util.HashMap;
import java.util.Scanner;;

public class ColorBalls {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        Map<String, Integer> balls = new HashMap<String, Integer>();

        // Read input and store it in map
        String line = scanner.nextLine();
        while (!line.equals("End")) {
            String[] parts = line.split(": ");
            String color = parts[0];
            int count = Integer.parseInt(parts[1]);
            balls.put(color, count);
            line = scanner.nextLine();
        }

        String minColor = "";
        int minCount = Integer.MAX_VALUE;

        // Find the smallest color group
        for (Map.Entry<String, Integer> entry : balls.entrySet()) {
            if (entry.getValue() < minCount) {
                minCount = entry.getValue();
                minColor = entry.getKey();
            }
        }

        // Take all balls except for the smallest color group
        int minSum = 0;
        for (Map.Entry<String, Integer> entry : balls.entrySet()) {
            if (!entry.getKey().equals(minColor)) {
                minSum += entry.getValue();
            }
        }

        System.out.println("Minimum number of balls: " + minSum);
    }
}
